src/main/java/com/hackoeur/jglm/Mat.java

```diff
@@ -41,14 +41,14 @@
  * 
  * @author James Royalty
  */
-public interface Mat {
+public interface Mat<T extends Vec> {
```

Comment:
@Zubnix: I would prefer not to have to change the signature of the interface for this patch, breaking compatibility for others.